### PR TITLE
Make bgzip -d work with non-.gz extensions (and minor nearby tidy ups)

### PR DESCRIPTION
**Pull request by jmarshall**
Without commenting on whether `.bgz` is an extension that should be encouraged :smile:, this fixes the usability problem in #129 by making the following work:

```
bgzip -d gnomad.exomes.r2.1.1.sites.vcf.bgz
```

As noted on that issue, `tabix gnomad.exomes.r2.1.1.sites.vcf.bgz 1:12000-13000`  already works fine.

Also various minor tidy ups nearby that were noticed in the course of doing this.

### CODE REVIEW

File: bgzip.c [hunk 1]
```diff
@@ -312,8 +296,17 @@ int main(int argc, char **argv)
             }
             else {
                 const int wrflags = O_WRONLY | O_CREAT | O_TRUNC;
+                char *name = argv[optind];
```
Comment (daviesrob) [inline bgzip.c]: Is there a danger here that if you tried to `bgzip -d -f something.bam.csi` then you might overwrite `something.bam` unintentionally?  Or given that it's necessary to use the `-f` option for this to happen, do we not worry as it was explicitly requested?

Comment (jmarshall) [reply bgzip.c]: Good point, I guess that is a new danger. I guess the possible answers are

1. Whitelist known compression extensions, namely `.gz` and `.bgz` (and …?)

2. Blacklist known index extensions, e.g., `.bai`, `.csi`, etc

3. Accept any extensions like this code does, and depend on the `-f` failsafe; if they say yes to that, they really do mean it.

4. Status quo, which is basically "only `.gz` is whitelisted"

Hmmm… option 3 and change to one of the others if problems ensue?
I guess (4) suggests that (1) is the minimal change…

Comment (daviesrob) [reply bgzip.c]: Sticking to `.gz` and `.bgz` would be safest.  Or we could make it politely refuse if the output file exists even if you've used `-f` _unless_ one of the known-safe extensions is used (and allow something like `-f -f` for "I really really want to do this").

Comment (jmarshall) [reply bgzip.c]: PR updated to do (1) essentially — asks you both “are you sure about this funny extension” and “are you sure about overwriting” as appropriate, and both questions can be suppressed with enough `-f -f`.

**Comment by jmarshall** [comment]
Thanks for the review — updated accordingly.

Also added one more nearby bug fix. It's a doozy!

The code currently works due to a superposition of two pieces of luck and the pooled allocator not memsetting the newly `pool_free()`ed block to garbage. The bug fix is needed for an upcoming PR (to implement `@SQ-AN`; um, hope nobody else has been working on that :thinking:).

**Comment by daviesrob** [comment]
As far as I know, no-one is looking at `@SQ-AN`, so feel free to carry on.

**Comment by daviesrob** [comment]
Fixes now merged, thanks for all the hard work :smiley:

**Pull request merged**